addons/sms/static/tests/sms_widget_test.js

```diff
@@ -1,128 +1,202 @@
+        const record = model.endsWith('partner') ? partnerRecord : visitorRecord;
+
+        const startParams = {serverData: {views}};
+        if (mockRPC) {
+            startParams.mockRPC = mockRPC;
+        }
+        const { env, openView } = await start(startParams, { debug });
+        const openViewParams = {
+            res_model: resModel,
+            views: [[false, 'form']],
         };
-        setupViewRegistries();
-        this.target = getFixture();
+        if (!newRecord) {
+            openViewParams.res_id = record;
+        }
+        await openView(openViewParams);
+        return env;
     }
-}, function () {
 
-    QUnit.module('SmsWidget');
+    hooks.beforeEach(() => {
+        target = getFixture();
+    });
+
+    /**
+     * Format data to fit the displayed information on this widget
+     *
+     * @example (0, 0, 'Unicode') => '0 characters, fits in 0 SMS (Unicode)'
+     * @param {number} characterCount number of characters in terms of bytes (utf emote = 2 bytes)
+     * @param {number} smsCount number of sms messages the message should fit inside of
+     * @param {string} encoding either GSM7 or Unicode
+     * @returns {string} correctly-formated string that should be displayed by the widget
+     */
+     function getExpectedWidgetString(characterCount, smsCount, encoding = "GSM7") {
+        return `${characterCount} characters, fits in ${smsCount} SMS (${encoding}) `;
+    }
 
     QUnit.test('Sms widgets are correctly rendered', async function (assert) {
         assert.expect(9);
-        await makeView({
-            type: "form",
-            resModel: "partner",
-            serverData: { models: this.data },
-            arch: /* xml */ `<form><sheet><field name="message" widget="sms_widget"/></sheet></form>`,
-        });
+        await openTestView('fields.sms.emojis.partner', `<form><sheet><field name="message" widget="sms_widget"/></sheet></form>`);
 
-        assert.containsOnce(this.target, '.o_sms_count', "Should have a sms counter");
-        assert.strictEqual(this.target.querySelector('.o_sms_count').textContent, '0 characters, fits in 0 SMS (GSM7) ',
-            'Should be "0 characters, fits in 0 SMS (GSM7) " by default');
+        assert.containsOnce(target, '.o_sms_count', "Should have a sms counter");
+        assert.strictEqual(target.querySelector('.o_sms_count').textContent, getExpectedWidgetString(0, 0));
         // GSM-7
-        await testUtils.fields.editAndTrigger(this.target.querySelector('.o_input'), "Hello from Odoo", 'input');
-        assert.strictEqual(this.target.querySelector('.o_sms_count').textContent, '15 characters, fits in 1 SMS (GSM7) ',
-            'Should be "15 characters, fits in 1 SMS (GSM7) " for "Hello from Odoo"');
+        await testUtils.fields.editAndTrigger(target.querySelector('.o_input'), "Hello from Odoo", 'input');
+        assert.strictEqual(target.querySelector('.o_sms_count').textContent, getExpectedWidgetString(15, 1));
```

Comment:
I keep thinking it's not useful to have a separate method to get the text.
As reading "15, 1", you don't know what is what, reading the text is much easier.

(If we ever change the text, the person changing that can be mad at me, I'll take full responsibility ^^)

Reply:
Alright then, regex-ed back to the original.

Tested the fields again, should work fine :)